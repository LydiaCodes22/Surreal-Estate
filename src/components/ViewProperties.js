import React from "react";

const ViewProperties = () => {
  return <h1>Properties</h1>;
};

export default ViewProperties;
