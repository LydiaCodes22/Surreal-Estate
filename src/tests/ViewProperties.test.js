import { render, screen } from "@testing-library/react";
import React from "react";
import ViewProperties from "../components/ViewProperties";

const validProperties = [
  {
    _id: "62b6199852cde50015266729",
    title: "10 bed castle",
    type: "big cottage",
    bedrooms: "7",
    bathrooms: "3",
    price: "1000000",
    city: "Leeds",
    email: "vee.uye@test.com",
  },
  {
    _id: "62b62b9841da260015adab2b",
    title: "test",
    city: "Sheffield",
    type: "Detached",
    bedrooms: "2",
    bathrooms: "2",
    price: "8",
    email: "test@test.com",
  },
];
describe("The view properties component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <ViewProperties properties={validProperties} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct property cards", () => {
    render(<ViewProperties properties={validProperties} />);
    expect(
      screen.getAllByRole("button", {
        name: /email/i,
      })
    ).toHaveLength(2);
    expect(screen.getByText("10 bed castle")).toBeInTheDocument();
    expect(screen.getByText("test")).toBeInTheDocument();
  });
});
