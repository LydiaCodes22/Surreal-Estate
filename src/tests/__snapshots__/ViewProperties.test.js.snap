// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The view properties component matches the snapshot 1`] = `
<DocumentFragment>
  <h1>
    Properties
  </h1>
  <div
    class="view-properties"
  >
    <div
      class="property-card"
    >
      <div>
        10 bed castle
      </div>
      <div>
        big cottage
      </div>
      <div>
        3
      </div>
      <div>
        7
      </div>
      <div>
        1000000
      </div>
      <div>
        Leeds
      </div>
      <div>
        <a
          href="mailto:vee.uye@test.com"
        >
          <button
            class="button"
            type="button"
          >
            Email
          </button>
        </a>
      </div>
    </div>
    <div
      class="property-card"
    >
      <div>
        test
      </div>
      <div>
        Detached
      </div>
      <div>
        2
      </div>
      <div>
        2
      </div>
      <div>
        8
      </div>
      <div>
        Sheffield
      </div>
      <div>
        <a
          href="mailto:test@test.com"
        >
          <button
            class="button"
            type="button"
          >
            Email
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
